extension/src/setup/studio.ts

```diff
@@ -134,9 +137,12 @@ export class Studio extends Disposable {
     verificationUrlWithParams.searchParams.append('code', userCode)
 
     await openUrl(verificationUrlWithParams.toString())
-    void waitForUriResponse('/studio-complete-auth', () => {
-      void this.requestStudioToken(deviceCode, tokenUri)
-    })
+    this.studioAccessTokenUriHandler = waitForUriResponse(
```

Comment:
Realized we aren't disposing the uri handler after use. I updated the code to dispose of the listener if:

* we get a response from Studio
* the user wants to try authentication from the start again